### ISSUE TITLE
Add support for a gba clone that uses a 3.2inch 320x480p screen 

### ISSUE THREAD

**Issue by JohnShatto** [DrUm78/FunKey-OS#25]
**Is your feature request related to a problem? Please describe.**
I have recently purchased a GBA clone that uses a modified version of your os. Its all in chinese and stuck on simplemenu which I do not prefer. No image you have seems to support the screen. They load but it is rotated 90 degrees to the right. 

**Describe the solution you'd like**
Maybe take this chinese os image and create a basic version of the gba mini that supports the higher 320x480 res screen. Since the gba mini image boots It hopefully wouldn't be too difficult. Its a perfect device for gba as its 2x the res of an OG gameboy advance. 

**Describe alternatives you've considered**
Learning chinese. So far I have figured out where shutdown is. So I have that going for me.

**Comment by DrUm78** [comment]
I usually do not port anything on a device I don't own, so unless you send me one...

**Comment by JohnShatto** [comment]
Not sure it needs a full port, seems like they just took your OS and translated the menus to Chinese, and some how got simplemenu loading on it. Anyway I could DD the SD card and send it your way where you can see what they have done? I backed it up since they don't provide any image for re imaging the card. When it boots it says FunKeyS SimUI X but I cant find anything with that boot logo. I have it working okay now just not english menus and stuck to a chinese translated simple menu. Tried going though the rootfs partition but dont know enough about the backend of the setup to figure it out yet. Going to keep cracking at it to see if I can just get it in english. So far when I modded the frontend file to be similar to the gba mini it booted to retrofe so they didnt delete the files just bypassed somehow. I cant send you one right now but will in the future if I cant get it. Thankfully they didn't change the menu beyond the language so muscle memory is working haha. 

I get not wanting to port stuff or mess with a device you don't have on hand. Its easily my favorite funkeyS based device. I really liked the gba mini but its more for taking with me places. Saves and games just swap between them easily so that's nice.  They even started to implement a brightness adjustment feature via the select and A button,  they had it messed up and it only went down so I adjusted it a bit to make it where select A is up and select B is down brightness. They have brightness not right as it wont go up all the way and shuts off the backlight completely at 0 so that's a bug. 

I don't expect miracles but if you could point me towards the files I need to mod to change it to English in the menus that would be good enough to make this less region locked to china. Where would you prefer I upload the file if you are willing to take a look?

**Comment by JohnShatto** [comment]
Update fixed the simple menu being in Chinese by swapping out the theme file with the 320x480p one from this Repo.
https://github.com/dkodr/SimUI/releases/tag/v1.2 
Now just the white quick menu you hail when you press the menu button is in Chinese.

**Comment by DrUm78** [comment]
I do not know SimUI at all so I cannot help. All I know is RetroFE and GMenu2X. Having the image won't help that much as I would rather need the source code. However, I can try to make my OS work on this device but I need a sample.

**Comment by JohnShatto** [comment]
That's fair I know you didn't make SimUI or SimpleMenu. I am pretty sure they adapted it due to resolution already being correct for the screen they choose. 
Found a document on the device in Chinese I translated it and the important bit here says.

"This firmware is based on the FunKeys version of Drum78. We would like to thank the FunKeys team and Drum78 for their 
contributions, and also thank the authors of SimpleMenu and SimUI for their hard work!
 This version mainly has the following updates
 :
 1. Removed the two front-ends RETROFE and GMenu2X, and used SimpleMenu with SimUI theme as 
the system interface, and Picoarch as the front-end of the simulator.
 2. SimpleMenu and Picoarch have been adapted and localized into Chinese. Most of them have been localized and support Chinese file names "

 So they took your Funkey OS and localized it for China.
 Mine had a broken menu button when I received it. It uses a touch sensor(piece of metal) that was badly soldered. Would you want one shipped directly from their store or should I order it to my place make sure its all good then ship it directly to you? How would I get in contact for a shipping address? I honestly think you will enjoy it. Thanks for being willing to try. 
 
 The V3s is a pretty great chip for GBA. Would be nice to see an offical version you make that uses RetroFE and GMenu2X (maybe RG351M theme since its the same resolution) where it isn't so piecemeal. Would you need the source code since its a V3s just like a Funkey? It boots all the funkey firmwares I have tried just has the screen all messed up and not the right direction.

**Comment by DrUm78** [comment]
@JohnShatto Thanks for the offer but not sure I want to develop on a 480x320 device. BTW what is the size of the device? Is it compact or much bigger than the FunKey S?

**Comment by JohnShatto** [comment]
It's the exact same size of a gameboy advance, so fullsize device not micro. It uses an oem shell. That's alright too bad you don't want to give it a go. I've just been using it with the stock interface on simpleUI and have the main game explore area in English just not the quick menu. Thankfully the quick menu is the same layout as your/funkey os. So it's not super bad being in Chinese. They are planning on making an SP version that uses oem sp shells so I am looking forward to giving that a try in thr future. The v3s chip is so quick to boot in 7 sec or so, a gameboy barely beets it booting up, resumes just like a funkey.

________________________________

**Comment by DrUm78** [comment]
Yeah I'm only interested in micro devices anyway so I'm closing the ticket, but thanks for the offer, appreciated. 👍
